src/Cryptol/ModuleSystem/Renamer/Monad.hs

```diff
@@ -250,10 +250,16 @@ checkEnv check (NamingEnv lenv) r rw0
 
   where
   newEnv         = NamingEnv newMap
-  (rwFin,newMap) = Map.mapAccumWithKey doNS rw0 lenv
+  (rwFin,newMap) = Map.mapAccumWithKey doNS rw0 lenv  -- lenv 1 ns at a time
   doNS rw ns     = Map.mapAccumWithKey (step ns) rw
 
-  step ns acc k xs = (acc', [head xs])
+  -- namespace, current state, k : parse name, xs : possible entities for k
```

Comment:
what's this change about?

Reply:
Apart from the comment, there was another bug here.   This code basically does some checking and adds the names to the environment.   If the names come from a bunch of imports (the "NoCheck" case), it is possible that names are ambiguous, and we are not reporting an error.  In that case, we should not arbitrarily pick the first possible resolution, but rather keep all the names, so that we can report an ambiguity error if one of the names is actually used.